diff --git a/CMakeLists.txt b/CMakeLists.txt
index a6f21cb..db39204 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -69,7 +69,7 @@ endif()
 
 include_directories(include)
 
-add_executable(flatc ${FlatBuffers_Compiler_SRCS})
+add_library(flatbuffer STATIC ${FlatBuffers_Compiler_SRCS})
 
 function(compile_flatbuffers_schema_to_cpp SRC_FBS)
   get_filename_component(SRC_FBS_DIR ${SRC_FBS} DIRECTORY)
@@ -93,7 +93,7 @@ endif()
 
 if(FLATBUFFERS_INSTALL)
   install(DIRECTORY include/flatbuffers DESTINATION include)
-  install(TARGETS flatc DESTINATION bin)
+  install(TARGETS flatbuffer DESTINATION lib)
 endif()
 
 if(FLATBUFFERS_BUILD_TESTS)
