diff --git a/lib/curl_config.h b/lib/curl_config.h
index 6acb5e4..e795bf3 100644
--- a/lib/curl_config.h
+++ b/lib/curl_config.h
@@ -199,8 +199,6 @@
 /* Define to 1 if the getaddrinfo function is threadsafe. */
 #define HAVE_GETADDRINFO_THREADSAFE 1
 
-/* Define to 1 if you have the `geteuid' function. */
-#define HAVE_GETEUID 1
 
 /* Define to 1 if you have the gethostbyaddr function. */
 #define HAVE_GETHOSTBYADDR 1
@@ -250,11 +248,6 @@
 /* Define to 1 if you have the `getprotobyname' function. */
 #define HAVE_GETPROTOBYNAME 1
 
-/* Define to 1 if you have the `getpwuid' function. */
-#define HAVE_GETPWUID 1
-
-/* Define to 1 if you have the `getpwuid_r' function. */
-#define HAVE_GETPWUID_R 1
 
 /* Define to 1 if you have the `getrlimit' function. */
 #define HAVE_GETRLIMIT 1
