diff --git a/Makefile b/Makefile
index 6e78f66..5deea66 100644
--- a/Makefile
+++ b/Makefile
@@ -38,7 +38,7 @@ RANLIB= ranlib
 # == END OF USER SETTINGS. NO NEED TO CHANGE ANYTHING BELOW THIS LINE =========
 
 # Convenience platforms targets.
-PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
+PLATS= aix ansi bsd freebsd generic linux tvos ios macosx mingw posix solaris
 
 # What to install.
 TO_BIN= lua luac
diff --git a/src/Makefile b/src/Makefile
index e4a3cd6..dca4284 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -6,7 +6,6 @@
 
 # Your platform. See PLATS for possible values.
 PLAT= none
-
 CC= gcc
 CFLAGS= -O2 -Wall $(MYCFLAGS)
 AR= ar rcu
@@ -98,8 +97,14 @@ generic:
 linux:
 	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl -lreadline -lhistory -lncurses"
 
+ios:
+	$(MAKE) all
+
+tvos:
+	$(MAKE) all
+
 macosx:
-	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-lreadline"
+	$(MAKE) all MYLIBS="-lreadline"
 # use this on Mac OS X 10.3-
 #	$(MAKE) all MYCFLAGS=-DLUA_USE_MACOSX
 
