diff --git a/src/luaconf.h b/src/luaconf.h
index 4c24bb1..c0df652 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -105,8 +105,8 @@
 #endif
 
 /* Note: changing the following defines breaks the Lua 5.1 ABI. */
-#define LUA_INTEGER	ptrdiff_t
-#define LUA_IDSIZE	60	/* Size of lua_Debug.short_src. */
+#define LUA_INTEGER ptrdiff_t
+#define LUA_IDSIZE  512 /* Size of lua_Debug.short_src. */
 /*
 ** Size of lauxlib and io.* on-stack buffers. Weird workaround to avoid using
 ** unreasonable amounts of stack space, but still retain ABI compatibility.
