diff --git a/src/Makefile b/src/Makefile
index 3f25192..de0eac5 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -168,7 +168,7 @@ else
   endif
   # Use Clang for OSX host.
   ifeq (Darwin,$(HOST_SYS))
-    DEFAULT_CC= clang
+    DEFAULT_CC= gcc
   endif
 endif
 
diff --git a/src/host/buildvm_asm.c b/src/host/buildvm_asm.c
index 28419c0..e9256b5 100644
--- a/src/host/buildvm_asm.c
+++ b/src/host/buildvm_asm.c
@@ -214,7 +214,8 @@ static void emit_asm_label(BuildCtx *ctx, const char *name, int size, int isfunc
   case BUILD_machasm:
     fprintf(ctx->fp,
       "\n\t.private_extern %s\n"
-      "%s:\n", name, name);
+      "\t.no_dead_strip %s\n"
+      "%s:\n", name, name, name);
     break;
   default:
     break;
diff --git a/src/luaconf.h b/src/luaconf.h
index 4c24bb1..c633fd6 100644
--- a/src/luaconf.h
+++ b/src/luaconf.h
@@ -106,7 +106,7 @@
 
 /* Note: changing the following defines breaks the Lua 5.1 ABI. */
 #define LUA_INTEGER	ptrdiff_t
-#define LUA_IDSIZE	60	/* Size of lua_Debug.short_src. */
+#define LUA_IDSIZE	512	/* Size of lua_Debug.short_src. */
 /*
 ** Size of lauxlib and io.* on-stack buffers. Weird workaround to avoid using
 ** unreasonable amounts of stack space, but still retain ABI compatibility.
