diff --git a/include/GLFW/glfw3native.h b/include/GLFW/glfw3native.h
index 267e75c..26bf782 100644
--- a/include/GLFW/glfw3native.h
+++ b/include/GLFW/glfw3native.h
@@ -94,7 +94,7 @@ extern "C" {
  #if defined(__OBJC__)
   #import <Cocoa/Cocoa.h>
  #else
-  #include <ApplicationServices/ApplicationServices.h>
+  //#include <ApplicationServices/ApplicationServices.h>
   typedef void* id;
  #endif
 #elif defined(GLFW_EXPOSE_NATIVE_X11) || defined(GLFW_EXPOSE_NATIVE_GLX)
@@ -200,7 +200,7 @@ GLFWAPI HGLRC glfwGetWGLContext(GLFWwindow* window);
  *
  *  @ingroup native
  */
-GLFWAPI CGDirectDisplayID glfwGetCocoaMonitor(GLFWmonitor* monitor);
+//GLFWAPI CGDirectDisplayID glfwGetCocoaMonitor(GLFWmonitor* monitor);

 /*! @brief Returns the `NSWindow` of the specified window.
  *
