From 0351d86e49d5c2be3863f9c867937aa5492f959b Mon Sep 17 00:00:00 2001
From: minggo <ming@cocos2d-x.org>
Date: Thu, 2 Nov 2017 10:11:44 +0800
Subject: [PATCH] macOS 10.12 patch

---
 src/cocoa_window.m | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cocoa_window.m b/src/cocoa_window.m
index b002e99..4ebd7d7 100644
--- a/src/cocoa_window.m
+++ b/src/cocoa_window.m
@@ -1004,11 +1004,11 @@ static GLFWbool createNativeWindow(_GLFWwindow* window,
     [window->ns.view setWantsBestResolutionOpenGLSurface:YES];
 #endif /*_GLFW_USE_RETINA*/
 
+    [window->ns.object setContentView:window->ns.view];
     [window->ns.object makeFirstResponder:window->ns.view];
     [window->ns.object setTitle:[NSString stringWithUTF8String:wndconfig->title]];
     [window->ns.object setDelegate:window->ns.delegate];
     [window->ns.object setAcceptsMouseMovedEvents:YES];
-    [window->ns.object setContentView:window->ns.view];
     [window->ns.object setRestorable:NO];
 
     return GLFW_TRUE;
-- 
2.6.2

