diff --git a/include/chipmunk/chipmunk.h b/include/chipmunk/chipmunk.h
index 616bfe0..0eb6eee 100644
--- a/include/chipmunk/chipmunk.h
+++ b/include/chipmunk/chipmunk.h
@@ -117,6 +117,7 @@ typedef struct cpSpace cpSpace;
 #include "cpConstraint.h"
 
 #include "cpSpace.h"
+#include "cpHastySpace.h"
 
 // Chipmunk 7.0.1
 #define CP_VERSION_MAJOR 7
diff --git a/include/chipmunk/chipmunk_types.h b/include/chipmunk/chipmunk_types.h
index 0816727..7aee0c7 100644
--- a/include/chipmunk/chipmunk_types.h
+++ b/include/chipmunk/chipmunk_types.h
@@ -26,35 +26,9 @@
 #include <float.h>
 #include <math.h>
 
-#ifdef __APPLE__
-   #include "TargetConditionals.h"
-#endif
-
-// Use CGTypes by default on iOS and Mac.
-// Also enables usage of doubles on 64 bit.
-// Performance is usually very comparable when the CPU cache is well utilised.
-#if (TARGET_OS_IPHONE || TARGET_OS_MAC) && (!defined CP_USE_CGTYPES)
-	#define CP_USE_CGTYPES 1
-#endif
-
-#if CP_USE_CGTYPES
-	#if TARGET_OS_IPHONE
-		#import <CoreGraphics/CGGeometry.h>
-		#import <CoreGraphics/CGAffineTransform.h>
-	#elif TARGET_OS_MAC
-		#include <ApplicationServices/ApplicationServices.h>
-	#endif
-	
-	#if defined(__LP64__) && __LP64__
-		#define CP_USE_DOUBLES 1
-	#else
-		#define CP_USE_DOUBLES 0
-	#endif
-#endif
-
 #ifndef CP_USE_DOUBLES
 	// Use doubles by default for higher precision.
-	#define CP_USE_DOUBLES 1
+	#define CP_USE_DOUBLES 0
 #endif
 
 /// @defgroup basicTypes Basic Types
diff --git a/src/cpHastySpace.c b/src/cpHastySpace.c
index 4e1bf2c..6f69ecc 100644
--- a/src/cpHastySpace.c
+++ b/src/cpHastySpace.c
@@ -6,7 +6,9 @@
 
 #include <pthread.h>
 //#include <sys/param.h >
+#ifdef __APPLE__
 #include <sys/sysctl.h>
+#endif
 
 #include "chipmunk/chipmunk_private.h"
 #include "chipmunk/cpHastySpace.h"
